nf_core/lint.py

```diff
@@ -508,22 +508,24 @@ def check_actions_branch_protection(self):
 
             # Check that the action is turned on for PRs to master
             try:
+                # Yaml 'on' parses as True - super weird
                 assert('master' in branchwf[True]['pull_request']['branches'])
             except (AssertionError, KeyError):
                 self.failed.append((5, "GitHub Actions 'branch' workflow should be triggered for PRs to master: '{}'".format(fn)))
             else:
                 self.passed.append((5, "GitHub Actions 'branch' workflow is triggered for PRs to master: '{}'".format(fn)))
 
             # Check that PRs are only ok if coming from an nf-core `dev` branch or a fork `patch` branch
-            PRMasterCheck = "{{ [[ $(git remote get-url origin) == *nf-core/{} ]] && [[ ${{GITHUB_HEAD_REF}} = \"dev\" ]]; }} || [[ ${{GITHUB_HEAD_REF}} == \"patch\" ]]".format(self.pipeline_name.lower())
-            steps = branchwf['jobs']['test']['steps']
-            try:
-                steps = branchwf['jobs']['test']['steps']
-                assert(any([PRMasterCheck in step.get('run', []) for step in steps]))
-            except (AssertionError, KeyError):
-                self.failed.append((5, "GitHub Actions 'branch' workflow should check that forks don't submit PRs to master: '{}'".format(fn)))
+            steps = branchwf.get('jobs', {}).get('test', {}).get('steps', [])
+            for step in steps:
+                has_name = step.get('name', '').strip() == 'Check PRs'
+                has_if = step.get('if', '').strip() == "github.repository == 'nf-core/{}'".format(self.pipeline_name.lower())
+                has_run = step.get('run', '').strip() == '{{ [[ $(git remote get-url origin) == *nf-core/{} ]] && [[ $GITHUB_HEAD_REF = "dev" ]]; }} || [[ $GITHUB_HEAD_REF == "patch" ]]'.format(self.pipeline_name.lower())
+                if has_name and has_if and has_run:
+                    self.passed.append((5, "GitHub Actions 'branch' workflow checks that forks don't submit PRs to master: '{}'".format(fn)))
+                    break
             else:
```

Comment:
Ah cool, I did not know of the for/else loop! (Learning by reviewing...)

Reply:
Yeah, it's a bit of a random pattern - I'm not aware of it in other languages. It can be nice for scenarios like this!